crates/sui-indexer/migrations/2023-10-06-204335_tx_indices/up.sql

```diff
@@ -0,0 +1,47 @@
+CREATE TABLE tx_senders (
+    cp_sequence_number          BIGINT       NOT NULL,
+    tx_sequence_number          BIGINT       NOT NULL,
+    -- SuiAddress in bytes.
+    sender                      BYTEA        NOT NULL,
+    PRIMARY KEY(sender, tx_sequence_number, cp_sequence_number)
+);
+CREATE INDEX tx_senders_tx_sequence_number_index ON tx_senders (tx_sequence_number ASC, cp_sequence_number ASC);
+
+CREATE TABLE tx_recipients (
+    cp_sequence_number          BIGINT       NOT NULL,
+    tx_sequence_number          BIGINT       NOT NULL,
+    -- SuiAddress in bytes.
+    recipient                   BYTEA        NOT NULL,
+    PRIMARY KEY(recipient, tx_sequence_number, cp_sequence_number)
+);
+CREATE INDEX tx_recipients_tx_sequence_number_index ON tx_recipients (tx_sequence_number ASC, cp_sequence_number ASC);
+
+CREATE TABLE tx_input_objects (
+    cp_sequence_number          BIGINT       NOT NULL,
+    tx_sequence_number          BIGINT       NOT NULL,
+    -- Object ID in bytes. 
+    object_id                   BYTEA        NOT NULL,
+    PRIMARY KEY(object_id, tx_sequence_number, cp_sequence_number)
```

Comment:
q: how come some of these `tx_` tables have an index, and others don't?

Reply:
those indices were added for tx_reader, others count on the primary key.